notebooks/Notebook2.md

```diff
@@ -18,17 +31,49 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+I spent a good chunk of this week adapting my Python code into Scala, adding
+some richness to it that takes advantage of Scala's DSL-building features. I
+got pretty stuck while trying to implement a "do for all players" function that
+has a more readable syntax than just using a for loop. The problem is that it's
+very difficult to have the iterator variable accessible in the body of the loop,
```

Comment:
I don't yet fully know the syntax you are using, but I have an idea for you to try out. Let's chat!